example/src/main.rs

```diff
@@ -34,63 +33,70 @@ use solaris::wei;
 use solaris::convert;
 
 #[cfg(test)]
-use solaris::{U256, Address};
+use types::{Address, H256, U256};
 
 #[test]
 fn badge_reg_test_fee() {
     let (mut evm, contract) = setup();
-    let reg = contract.functions();
 
     // Initial fee is 1 ETH
-    assert_eq!(U256::from(reg.fee().call(&mut evm).unwrap()), wei::from_ether(1));
+    assert_eq!(
+        evm.call(contract.functions().fee()).unwrap(),
+        wei::from_ether(1)
+    );
 
     // The owner should be able to set the fee
-    reg.set_fee().transact(wei::from_gwei(10), &mut evm).unwrap();
+    evm.call(contract.functions().set_fee(wei::from_gwei(10)))
+        .unwrap();
 
     // Fee should be updated
-    assert_eq!(U256::from(reg.fee().call(&mut evm).unwrap()), wei::from_gwei(10));
+    assert_eq!(
+        evm.call(contract.functions().fee()).unwrap(),
+        wei::from_gwei(10)
+    );
 
     // Other address should not be allowed to change the fee
-    evm.with_sender(10.into());
-    reg.set_fee().transact(wei::from_gwei(10), &mut evm).unwrap_err();
+    evm.with_sender(10.into())
+        .transact(contract.functions().set_fee(wei::from_gwei(10)))
```

Comment:
Minor nitpick: i would suggest putting some different number here, just to avoid a possible problem with the contract no-oping if the values match.

Also, how exactly do you check that the `set_fee` execution fails in this step?

Reply:
(Sorry if this is not completely relevant to the more major problems of the PR)

Reply:
thanks for taking a look @kirushik : )

> i would suggest putting some different number here, just to avoid a possible problem with the contract no-oping if the values match.

do you mean to use 10 as the sender and 5 as the gwei for example?

> Also, how exactly do you check that the set_fee execution fails in this step?

there's an `unwrap` below that's not changed and therefore not visible in the diff. does that answer the question?